src/Solus.jl

```diff
@@ -0,0 +1,83 @@
+module Solus
+
+using Distributions, Statistics, LinearAlgebra, DocStringExtensions
+
+include("spaces.jl")
+
+"""
+    SolusProblem
+
+An uncertainty quantification problem.
+
+# Fields
+
+$(DocStringExtensions.FIELDS)
+"""
+struct SolusProblem{P<:MvNormal,M,O,S<:HilbertSpace}
+    """
+    The prior distribution for the parameters ``θ``. Currently only `MvNormal` objects are supported.
+    """
+    prior::P
+
+    """
+    A forward model maps the parameter ``θ`` to a predicted observation.
+    """
+    forwardmodel::M
+    
+    """
+    The observed data.
+    """
+    obs::O
+
+    """
+    The space on which the forward model output and observation exist. `DefaultSpace()` is the default.
+    """
+    space::S
+end
+SolusProblem(prior, forwardmodel, obs) = SolusProblem(prior, forwardmodel, obs, DefaultSpace())
+
+
+"""
+    Ensemble
+
+An ensemble of `inputs` and their corresponding `outputs` from the forward model.
+
+# Fields
+
+$(DocStringExtensions.FIELDS)
+"""
+struct Ensemble{T,O}
+    """
+    A matrix of inputs: each column corresponds to a vector of ``θ``s
+    """
+    inputs::Matrix{T}
+    """
+    Result of forward model for each column of `inputs`.
+    """
+    outputs::Vector{O}
+end
+
+"""
+    neki_iter(prob::SolusProblem, ens::Ensemble)
+
+Peform an iteration of NEKI, returning a new `Ensemble` object.
+"""
+function neki_iter(prob::SolusProblem, ens::Ensemble)
```

Comment:
Alfredo et al. call this the ensemble Kalman flow (though EKF invites confusion with the filter).